src/main/groovy/edu/oregonstate/mist/slashcourses/resources/SlashCourseResource.groovy

```diff
@@ -134,4 +142,51 @@ class SlashCourseResource extends Resource {
         }
         returnResponse
     }
+
+    /**
+     * Respond to PUT requests by updating a course object according to course CRN.
+     * If the course object doesn't exist, create one with POST requests.
+     *
+     * @param crn
+     * @param newCourse
+     * @return response containing the result or error message
+     */
+    @PUT
+    @Path('{crn: \\d+}')
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response putByCRN (@PathParam('crn') Integer crn, SlashCourse newCourse) {
+        Response returnResponse
+        SlashCourse checkForCourseCRN = slashCourseDAO.getByCRN(crn)
+
+        // if the course object doesn't exist, call postCourse() to create one
+        if (!checkForCourseCRN) {
+            postCourse(newCourse)
+            URI createdURI = URI.create("/" + instructorDAO.getLatestInstructorId())
+            returnResponse = Response.created(createdURI).build()
+        } else {
+            // courseNum, CourseName and slash must not be null
+            String newCourseNum      = Optional.fromNullable(newCourse.courseNum).or(checkForCourseCRN.courseNum)
```

Comment:
Can you use the groovy "with" clause to make this code a bit shorter: https://dzone.com/articles/getting-groovy-with-with ?